<?xml version="1.0"?>
<launch>
  <arg name="force_or_potential" default="True" /> 
  <!--<arg name="gv_or_mav" />  -->


  <arg name="cmd_vel_topic" value="/cmd_vel" /> 
  <!--<arg name="pose_topic" value="/odometry"/>-->  
  <arg name="pose_topic" value="/ground_truth/state" /> 
  <arg name="laser_scan_topic" value="/scan" />


 <!-- Run the slave controller  -->
 <node name="slave_controller" pkg="haptic_teleoperation" type="controller_ros" ns="/haptic_teleoperation">
    <rosparam command="load" file="$(find haptic_teleoperation)/config/slave_controller_params.yaml" />
    <rosparam command="load" file="$(find haptic_teleoperation)/config/slave_params.yaml" />
    <rosparam command="load" file="$(find haptic_teleoperation)/config/master_params.yaml" />
    <remap from="/cmd_vel" to="$(arg cmd_vel_topic)"/> 
    <remap from="/pose" to="$(arg pose_topic)"/>  
  </node>

 <!-- Run the master controller controller  -->
  <node name="master_controller" pkg="haptic_teleoperation" type="controller_ros" ns="/haptic_teleoperation" >
    <rosparam command="load" file="$(find haptic_teleoperation)/config/slave_params.yaml" />
    <rosparam command="load" file="$(find haptic_teleoperation)/config/master_controller_params.yaml" />
    <rosparam command="load" file="$(find haptic_teleoperation)/config/master_params.yaml" />
    <remap from="/pose" to="$(arg pose_topic)"/>  
  </node>

 <!-- Convert the laser data into point clout data (subscribe for lasersacn and publish point cloud)-->
  <node name="laser2pointcloud" pkg="haptic_teleoperation" type="laser2pointcloud" ns="/haptic_teleoperation" >
	<remap from="/laserscan" to="$(arg laser_scan_topic)"/>
        <remap from="/base_link" to="laser0_frame"/>
  	
  </node>
 
<group if="$(arg force_or_potential)">

 <!-- Run the potential field node that calculated the potential field around the robot ( find the risk vector)-->
  <node name="potential_field" pkg="haptic_teleoperation" type="potential_field" output="screen" ns="/haptic_teleoperation">
    <rosparam command="load" file="$(find haptic_teleoperation)/config/potential_field.yaml" />
    <remap from="pose" to="$(arg pose_topic)"/> 
  </node>
</group>
<group unless="$(arg force_or_potential)">
 <!-- Run the force field node that find the envirnmental force that acts on the the robot -->
<node name="force_field" pkg="haptic_teleoperation" type="force_field"  ns="/haptic_teleoperation">
    <rosparam command="load" file="$(find haptic_teleoperation)/config/force_field_params.yaml" />
    <remap from="/base_link" to="laser0_frame"/> 
  </node>

</group>
 
<!--<node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>-->

</launch>
